Python/bytecodes.c

```diff
@@ -3563,6 +3563,7 @@ dummy_func(
             PyFunctionObject *init = (PyFunctionObject *)cls->_spec_cache.init;
             PyCodeObject *code = (PyCodeObject *)init->func_code;
             DEOPT_IF(code->co_argcount != oparg+1);
+            DEOPT_IF(_Py_Specialize_CheckCode(code) != _Py_Specialize_Ok);
```

Comment:
We want guards to be fast; just a few machine instructions if possible.
Even if the C compiler can inline and optimize it, I'd prefer the stripped down version:
```suggestion
            DEOPT_IF(code->co_flags & (CO_VARKEYWORDS | CO_VARARGS | CO_OPTIMIZED)) != CO_OPTIMIZED)
            DEOPT_IF(code->co_kwonlyargcount);
```